app/models/invite.rb

```diff
@@ -13,12 +19,15 @@ class UserExists < StandardError; end
   belongs_to :topic
   belongs_to :invited_by, class_name: 'User'
 
+  has_many :invited_users
+  has_many :users, through: :invited_users
   has_many :invited_groups
   has_many :groups, through: :invited_groups
   has_many :topic_invites
   has_many :topics, through: :topic_invites, source: :topic
   validates_presence_of :invited_by_id
   validates :email, email: true, format: { with: EmailValidator.email_regex }, allow_blank: true
+  validates :max_redemptions_allowed, numericality: { only_integer: true, greater_than: 0 }
```

Comment:
We need a custom validation here too. If it is a multi-invite link, it has to be greater than 2 and smaller than the site setting.